pkg/udf/rpc/tracker.go

```diff
@@ -0,0 +1,75 @@
+package rpc
+
+import (
+	"log"
+	"sync"
+
+	"github.com/numaproj/numaflow/pkg/isb"
+)
+
+type Tracker struct {
+	lock sync.RWMutex
+	m    map[string]*isb.ReadMessage
+}
+
+func NewTracker() *Tracker {
+	return &Tracker{
+		m: make(map[string]*isb.ReadMessage),
+	}
+}
+
+func (t *Tracker) AddRequest(msg *isb.ReadMessage) {
+	id := msg.ReadOffset.String()
```

Comment:
lock

Reply:
I have kept the lock inside the `t.set` function. Do you see a need to lock here instead?

Reply:
Forgot to remove the comment.